### PR TITLE
Search improvements

### PR DESCRIPTION
**Pull request by philippfromme**
Improves search by

* sorting matches by location (when searching for "foo", "foo bar" will appear before "bar foo")

brave_noS564awP1

Depends on https://github.com/bpmn-io/diagram-js/pull/913

Closes https://github.com/bpmn-io/bpmn-js/issues/2189

### CODE REVIEW

**Comment by barmac** [comment]
@philippfromme There are conflicts to resolve in this PR

File: lib/features/search/BpmnSearchProvider.js [hunk 1]
```diff
@@ -48,48 +45,114 @@ BpmnSearchProvider.prototype.find = function(pattern) {
...
-    return element !== rootElement;
-  });
-
-  elements = map(elements, function(element) {
-    return {
-      primaryTokens: matchAndSplit(getLabel(element), pattern),
-      secondaryTokens: matchAndSplit(element.id, pattern),
-      element: element
-    };
+    return !isLabel(element) && element !== rootElement;
```
Comment (barmac) [inline lib/features/search/BpmnSearchProvider.js:48]: 👍 for helper usage

**Comment by barmac** [comment]
Let's fix the conflict, and it is ready to be merged.

**Comment by nikku** [comment]
I checked this PR once again, UX wise, as I previously lost one of my reviews :see_no_evil: 

One thing I found is that clicking anywhere on the canvas with search open would reset the viewport to the previous viewbox. I believe if the user clicks anywhere (closing search), then they want to commit the current viewbox, and do not expect it to "jump back". Things are, of course, different if they press `ESC` ("GET ME OUT OF THERE").

Also I found that sometimes focussing of elements is odd (cf. jumping when selecting different sub-processes that would actually be "in view". 

Both visualized in the following recording:


capture ngU8IG_optimized

**Review by nikku — CHANGES_REQUESTED** [review]
Cf. https://github.com/bpmn-io/bpmn-js/pull/2187#issuecomment-2303198593.

Let's consider if we want to fix the last bug.

**Review by abdul99ahad — DISMISSED** [review]
Merge conflicts in the PR. Please resolve them.

**Review by nikku — APPROVED** [review]
Great stuff.

**Comment by nikku** [comment]
Updated on top of latest DEV branch + pulled in diagram-js@14.10.0.

**Comment by nikku** [comment]
Did some integration testing on the [released changes](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#17110), and identified potential follow-ups:

* https://github.com/bpmn-io/bpmn-js/issues/2232
* https://github.com/bpmn-io/bpmn-js/issues/2233

**Pull request merged**